### ISSUE TITLE
Add better validation for Preferences UI Items

### ISSUE THREAD

**Issue by latenitefilms** [fcpxhacks/fcpxhacks#416]

**Comment by latenitefilms** [comment]
I'll tackle this once #306 is merged into develop.

**Comment by latenitefilms** [comment]
@randomeizer - Just in case you decide to address this in #306 before merging into `develop`, what I'm thinking, for example with a checkbox, is basically adding JavaScript to the `cp.web.generate` checkbox generator that automatically disables the checkbox as soon as it's clicked. It's then up to the plugin to "inject" JavaScript back in to re-enable it with the correct value. We could probably do this as a callback.

**Comment by latenitefilms** [comment]
FYI: Will tackle AFTER #443 is complete.

**Comment by latenitefilms** [comment]
Completed in #434.
